modules/decklinksrc/decklinksrc.py

```diff
@@ -1,40 +1,42 @@
 import cv2 as cv2
 import numpy as np
+import threading
+
 
 class DeckLinkSRC:
 
-    def __init__(self):
+    def __init__(self, videoPipeline):
         self.__currentFrame = None
-        self.capture = cv2.VideoCapture('decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink') #Starts capture on initialization of object
-        #Because of this, we no longer need start stream code from Aryan
+        self.capture = cv2.VideoCapture(
+            'decklinkvideosrc mode=7 connection=0 ! videoconvert ! appsink')  # Starts capture on initialization of object
+        self.videoPipeline = videoPipeline
+        main = threading.Thread(target=self._main_)
+        main.start()
+
+    def _main_(self):
+        self.videoPipeline.addNewPackage(self.grab())
 
-    def stop(self): #Logic for stopping video feed by releasing capture and destroying any windows open.
+    def stop(self):  # Logic for stopping video feed by releasing capture and destroying any windows open.
         self.capture.release()
         cv2.destroyAllWindows()
-    
 
-    def grab(self, frame): #Logic for grabbing frame from deckLink
-        self.cv2.imshow('Grabbedframe', frame)
-    
+    def grab(self):  # Logic for displaying single frame from deckLink
+        self.__currentFrame = self.capture.read()
+        return self.get_frame()
 
-    def grab(self): #Logic for displaying single frame from deckLink
-        cv2.imshow('Grabbedframe', self.__currentFrame)
-    
-    def get_frame(self): #Logic for returning the current frame as a numpy array
+    def get_frame(self):  # Logic for returning the current frame as a numpy array
         return self.__currentFrame
-    
 
     def quitProgram(self):
         self.capture.release()
         cv2.destroyAllWindows()
 
-
     # We need the file name, x dimensions, and y dimensions
     def recordVideo(self, filename, xdim, ydim):
```

Comment:
This function doesn't seem to exit until the camera is no longer capturing any data - maybe it needs a new thread?

Reply:
I mean to say that the `recordVideo()` function is going to be constantly saving video, so the thread designated for this module will spend all its time saving video, up until the camera stream is shut off. We want it so that the video is saving simultaneously as other processes are happening (e.g. sending the video stream to another module). So I was saying maybe this video saver needs its own thread to enable that multitasking?

Reply:
You know what, I don't even think we need this functionality, I thought you were talking about quitProgram but frankly I don't see the value in refactoring something we're never gonna use anyways so I'll just drop it.